mantis-runtime/src/main/java/io/mantisrx/runtime/source/Source.java

```diff
@@ -18,13 +18,14 @@
 
 import io.mantisrx.runtime.Context;
 import io.mantisrx.runtime.parameter.ParameterDefinition;
+import java.io.Closeable;
 import java.util.Collections;
 import java.util.List;
 import rx.Observable;
 import rx.functions.Func2;
 
 
-public interface Source<T> extends Func2<Context, Index, Observable<Observable<T>>> {
+public interface Source<T> extends Func2<Context, Index, Observable<Observable<T>>>, Closeable {
```

Comment:
we could avoid all the empty `close()` calls in connectors and user code by having a default empty `close()` impl in the `Source` interface

Reply:
Hmm, the decision to avoid the default close() implementation in the source API was deliberate. We want to ensure that users who implement these APIs think about what resources they need to close actively.

Reply:
For a new interface, I agree. But for an existing interface with existing implementations which has been around for a while, it is a tradeoff between a user's build breaking to add an empty `close()` v/s keeping that as the default behavior with an option to override.

Reply:
I don't think this should break a user's build because we are doing a major version upgrade (2.0.0) with the new Titus set of changes. nfmantis modules that pull in mantis artifacts will also undergo a major upgrade. This will ensure only developers that opt in for the new version will receive it. We will also have a bom release reflecting this.

Reply:
this makes sense if it is part of 2.0.0